app/com/arpnetworking/rollups/RollupPeriod.java

```diff
@@ -92,6 +95,23 @@ public SamplingUnit getSamplingUnit() {
         _samplingUnit = samplingUnit;
     }
 
+    /**
+     * Get the next smallest rollup period, if any.
+     *
+     * @return An {@code Optional} containing the next smallest RollupPeriod, or {@link Optional#empty()}
+     * if this is already the smallest.
+     */
+    public Optional<RollupPeriod> nextSmallest() {
+        final int i = this.ordinal();
+        if (i == 0) {
+            return Optional.empty();
+        }
+        return Optional.of(VALUES.get((i - 1) % VALUES.size()));
+    }
+
+    // values() will create a new array on every call to nextSmallest without this.
+    private static final List<RollupPeriod> VALUES = ImmutableList.copyOf(values());
```

Comment:
I mean if you want to get picky, we can go from O(N) to O(1) by computing the pair-wise "next smallest" in a `static {}` block. This is a regular pattern we've used with enumerations.

Reply:
`nextSmallest` is already O(1) as implemented.

I think you're suggesting it an alternative would be a lookup table or additional field per-instance?

Reply:
Yep